__test__/users.test.ts

```diff
@@ -0,0 +1,263 @@
+    const body = {
+      id: "54",
+      first_name: "John",
+      last_name: "John",
+      email: "john@gmail.com",
+      role: "Student",
+      address: "123 Main Street",
+      phone_number: "1234567890",
+    };
+
+    await makeHTTPRequest(
+      userHandler,
+      "/api/users/",
+      undefined,
+      "POST",
+      body,
+      StatusCodes.INTERNAL_SERVER_ERROR,
+      INTERNAL_SERVER_ERROR
+    );
+  });
+
+  it("creates an Admin user", async () => {
+    const body = {
+      id: "50",
+      first_name: "Admin",
+      last_name: "Doe",
+      email: "newAdmin@gmail.com",
+      role: "Admin",
+      address: "123 Main Street",
+      phone_number: "1234567890",
+    };
+    await makeHTTPRequest(
+      userHandler,
+      "/api/users/",
+      undefined,
+      "POST",
+      body,
+      StatusCodes.CREATED,
+      body
+    );
+  });
+
+  it("creates an Teacher user", async () => {
+    const body = {
+      id: "45",
+      first_name: "Teacher",
+      last_name: "Doe",
+      email: "newTeacher@gmail.com",
+      role: "Admin",
+      address: "123 Main Street",
+      phone_number: "1234567890",
+    };
+    await makeHTTPRequest(
+      userHandler,
+      "/api/users/",
+      undefined,
+      "POST",
+      body,
+      StatusCodes.CREATED,
+      body
+    );
+  });
+});
+
+describe("[GET] /api/users/[id]", () => {
+  it("look for a user that exists", async () => {
+    const expected: User = {
+      id: "1",
+      first_name: "John",
+      last_name: "Doe",
+      email: "john@gmail.com",
+      role: "Student",
+      address: "123 Main Street",
+      phone_number: "1234567890",
+    };
+
+    const query = {
+      id: 1,
+    };
+
+    await makeHTTPRequest(
+      userIDHandler,
+      "/api/users/1",
+      query,
+      "GET",
+      undefined,
+      StatusCodes.ACCEPTED,
+      expected
+    );
+  });
+
+  it("fails for a user that does not exist", async () => {
+    const query = {
+      id: 101,
+    };
+
+    await makeHTTPRequest(
+      userIDHandler,
+      "/api/users/101",
+      query,
+      "GET",
+      undefined,
+      404,
+      USER_NOT_FOUND_ERROR
+    );
+  });
+});
+
+describe("[PATCH] /api/users/[id]", () => {
+  it("editing everything for a user that does exist", async () => {
+    const expected: User = {
+      id: "1",
+      first_name: "Bill",
+      last_name: "Brown",
+      email: "john123@gmail.com",
+      role: "Admin",
+      address: "456 Main Street",
+      phone_number: "4567890",
+    };
+
+    const query = {
+      id: 1,
+      
+    };
+
+    const body = {
+      first_name: "Bill",
+      last_name: "Brown",
+      email: "john123@gmail.com",
+      role: "Admin",
+      address: "456 Main Street",
+      phone_number: "4567890",
+    }
+
+    await makeHTTPRequest(
+      userIDHandler,
+      "/api/users/1",
+      query,
+      "PATCH",
+      body,
+      StatusCodes.CREATED,
+      expected
+    );
+  });
+
```

Comment:
can we add a test that tests only providing some of the inputs within the body?

Reply:
done